simpeg/potential_fields/gravity/simulation.py

```diff
@@ -282,18 +282,45 @@ def getJtJdiag(self, m, W=None, f=None):
             else np.ones(self.survey.nD, dtype=np.float64)
         )
 
-        # Compute diagonal of G.T @ G
+        # Compute gtg (G.T @ W.T @ W @ G) if it's not cached, or if the
+        # weights are not the same.
```

Comment:
Should we move this to the docstrings?

Reply:
Yes, I agree. I'll probably add it under the Notes section.